src/oscar/templatetags/category_tags.py

```diff
@@ -3,10 +3,50 @@
 from oscar.core.loading import get_model
 
 register = template.Library()
-Category = get_model('catalogue', 'category')
+Category = get_model("catalogue", "category")
 
 
-@register.simple_tag(name="category_tree")
+class CheapCategoryInfo(dict):
+    """
+    Wrapper class for Category.
+
+    Besides allowing inclusion of extra info, useful while rendering a template,
+    this class hides any expensive properties people should not use by accident
+    in templates.
+
+    This replaces both the node as the info object returned by the ``category_tree``
+    templatetag, so it mimics a tuple of 2 items (which are the same) for
+    backwards compatibility.
+    """
+
+    def __init__(self, category, **info):
+        super(CheapCategoryInfo, self).__init__(info)
+        self.category = category
+        self.field_names = set(
+            [field.name for field in category._meta.get_fields(include_parents=True)]
+        )
+
+    def get_absolute_url(self):
+        return self["url"]
+
+    def __getattr__(self, name):
+        "Only allow access to cheap direct attributes of ``Category``"
+        if name in self.field_names:
+            return getattr(self.category, name)
+
```

Comment:
As most attributes won't be used ever, this should be just fine. We could set the attribute on the instance, so the next access will be local. 

```python
def __getattr__(self, name):
    "Only allow access to cheap direct attributes of ``Category``"
    if name in self.field_names:
        value = getattr(self.category, name)
        setattr(self, name, value)
        return value

```

Reply:
Ok I added a metaclass that sets up these accessors at type creation time.